src/librustc/session/filesearch.rs

```diff
@@ -93,29 +74,30 @@ impl<'a> FileSearch<'a> {
                     }
                 }
             }
-        });
+        }
     }
 
     pub fn new(sysroot: &'a Path,
                triple: &'a str,
-               search_paths: &'a SearchPaths,
-               kind: PathKind) -> FileSearch<'a> {
+               search_paths: &'a Vec<SearchPath>,
+               tlib_path: &'a SearchPath,
+               kind: PathKind)
+               -> FileSearch<'a> {
         debug!("using sysroot = {}, triple = {}", sysroot.display(), triple);
         FileSearch {
             sysroot,
-            search_paths,
             triple,
+            search_paths,
+            tlib_path,
             kind,
         }
     }
 
     // Returns a list of directories where target-specific dylibs might be located.
     pub fn get_dylib_search_paths(&self) -> Vec<PathBuf> {
-        let mut paths = Vec::new();
-        self.for_each_lib_search_path(|lib_search_path, _| {
-            paths.push(lib_search_path.to_path_buf());
-        });
-        paths
+        self.search_paths()
+            .map(|sp| sp.dir.to_path_buf())
+            .collect()
```

Comment:
This method makes me suspicious, couldn't it be `-> impl Iterator<Item = &'a Path>`?

Reply:
There's a single call site for this method. That call site fiddles with the result in a way that is a bit simpler with a `Vec` than an iterator, so I don't think there's much to be gained from changing it.

Reply:
There's more than one call site - see https://github.com/rust-lang/rust/pull/56090#discussion_r235856904.